universal-application-tool-0.0.1/app/views/components/Icons.java

```diff
@@ -0,0 +1,67 @@
+package views.components;
+
+import j2html.tags.ContainerTag;
+import services.question.QuestionType;
+
+public class Icons {
+
+  public static final String ADDRESS_SVG_PATH =
+      "M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38"
+          + " 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z";
+  public static final String NAME_SVG_PATH =
+      "M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34"
+          + " 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99"
+          + " 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z";
+  public static final String SEARCH_SVG_PATH =
+      "M55.146,51.887L41.588,37.786c3.486-4.144,5.396-9.358,5.396-14.786c0-12.682-10.318-23-23-23s-23,10.318-23,23"
+          + "  s10.318,23,23,23c4.761,0,9.298-1.436,13.177-4.162l13.661,14.208c0.571,0.593,1.339,0.92,2.162,0.92"
+          + "  c0.779,0,1.518-0.297,2.079-0.837C56.255,54.982,56.293,53.08,55.146,51.887z"
+          + " M23.984,6c9.374,0,17,7.626,17,17s-7.626,17-17,17 "
+          + " s-17-7.626-17-17S14.61,6,23.984,6z";
+  public static final String TEXT_SVG_PATH =
+      "M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0"
+          + " 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4"
+          + " 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5"
+          + " 0-2.21-1.79-4-4-4z";
+
+  public static ContainerTag questionTypeSvg(QuestionType type, int size) {
+    return questionTypeSvg(type, size, size);
+  }
+
+  public static ContainerTag questionTypeSvg(QuestionType type, int width, int height) {
+    String iconPath = "";
+    switch (type) {
+      case ADDRESS:
+        iconPath = Icons.ADDRESS_SVG_PATH;
+        break;
+      case NAME:
+        iconPath = Icons.NAME_SVG_PATH;
+        break;
+      case TEXT:
+      default:
+        iconPath = Icons.TEXT_SVG_PATH;
+    }
+    return svg(iconPath, width, height);
+  }
+
+  public static ContainerTag svg(String pathString, int size) {
+    return svg(pathString, size, size);
+  }
+
+  public static ContainerTag svg(String pathString, int width, int height) {
+    return svg(pathString).attr("viewBox", String.format("0 0 %1$d %2$d", width, height));
+  }
+
+  public static ContainerTag svg(String pathString) {
+    return new ContainerTag("svg")
+        .attr("xmlns", "http://www.w3.org/2000/svg")
+        .attr("fill", "currentColor")
+        .attr("stroke", "currentColor")
+        .attr("stroke-width", "1%")
+        .attr("aria-hidden", "true")
+        .with(
+            new ContainerTag("path")
+                .attr("xmlns", "http://www.w3.org/2000/svg")
+                .attr("d", pathString));
+  }
+}
```

Comment:
praise: in the PR description you said this was ugly as sin but tbh it looks fairly similar to how several different React project I've seen handled SVGs.